contracts/TaskBook.sol

```diff
@@ -5,83 +5,158 @@ import './AccountManager.sol';
 contract TaskBook is AccountManager {
 
 	uint private numTasks = 0;
-	uint constant maxSolvers = 10;
-	uint constant maxChallengers = 10;
+	uint private forcedErrorThreshold = 42;
 
 	event TaskCreated(uint taskID, uint minDeposit, uint blockNumber);
 	event SolverSelected(uint indexed taskID, address solver, bytes32 taskData, uint minDeposit);
-	event SolutionCommitted(uint taskID, uint minDeposit, bytes32 taskData, address solver);
+	event SolutionsCommitted(uint taskID, uint minDeposit, bytes32 taskData, address solver);
+	event SolutionRevealed(uint taskID, uint randomBits);
+	event TaskStateChange(uint taskID, uint state);
 
-	mapping(uint => Task) private tasks;
-	mapping(address => mapping(uint => bytes32)) private solverRandomBitsHash;
-	mapping(address => mapping(uint => bytes32)) private solutions;
-	mapping(address => mapping(uint => bytes32)) private verifierIntent;
+	struct Challenge {
+		address challenger;
+		bytes32 intentHash;
+	}
 
 	struct Task {
 		address owner;
-		address[] solvers;
 		address selectedSolver;
 		uint minDeposit;
 		bytes32 taskData;
-		uint numSolvers;
-		address[] challengers;
-		uint numChallengers;
+		mapping(address => bytes32) challenges;
+		uint state;
 	}
 
+	struct Solution {
+		bytes32 solutionHash0;
+		bytes32 solutionHash1;
+		bool solution0Correct;
+		address[] solution0Challengers;
+		address[] solution1Challengers;
+	}
+
+	mapping(uint => Task) private tasks;
+	mapping(address => mapping(uint => bytes32)) private solverRandomBitsHash;
```

Comment:
`solverRandomBitsHash` can’t be a global mapping. What if a Solver wants to solve multiple tasks on Truebit. Their engagements with multiple tasks would over-write each other’s random bits hashes?

Reply:
Its a nested mapping so it would work. Solver address => taskID => randomBitsHash

Reply:
I see. That works!